experiment/template/eVOLVER_module.py

```diff
@@ -241,112 +287,147 @@ def initialize_exp(vials):
 
     if dpu_evolver_ns is None:
         print("Waiting for evolver connection...")
+        logger.info('waiting for eVOLVER connection')
 
     save_path = os.path.dirname(os.path.realpath(__file__))
-    dir_path = os.path.join(save_path,custom_script.EXP_NAME)
-
-
+    dir_path = os.path.join(save_path, EXP_NAME)
 
     while dpu_evolver_ns is None:
         pass
 
     if os.path.exists(dir_path):
-        exp_continue = input('Continue from existing experiment? (y/n): ')
+        logger.info('found an existing experiment')
+        if always_yes:
+            exp_continue = 'y'
+        else:
+            exp_continue = input('Continue from existing experiment? (y/n): ')
     else:
         exp_continue = 'n'
 
     if exp_continue == 'n':
 
         start_time = time.time()
         if os.path.exists(dir_path):
-            exp_overwrite = input('Directory aleady exists. Overwrite with new experiment? (y/n): ')
+            if always_yes:
+                exp_overwrite = 'y'
+            else:
+                exp_overwrite = input('Directory aleady exists. '
+                                      'Overwrite with new experiment? (y/n): ')
+            logger.info('data directory already exists')
             if exp_overwrite == 'y':
+                logger.info('deleting existing data directory')
                 shutil.rmtree(dir_path)
             else:
-                print('Change experiment name in custom_script.py and then restart...')
+                print('Change experiment name in custom_script.py '
+                      'and then restart...')
+                logger.warning('not deleting existing data directory, exiting')
                 exit() #exit
 
+        logger.debug('creating data directories')
         os.makedirs(os.path.join(dir_path,'OD'))
         os.makedirs(os.path.join(dir_path,'temp'))
         os.makedirs(os.path.join(dir_path,'temp_config'))
-        if custom_script.OPERATION_MODE == 'turbidostat':
+        if OPERATION_MODE == 'turbidostat':
```

Comment:
If you really want to make this super clean, make `turbidostat` and `chemostat` be final constants as well. Not really important though.

Reply:
I also realized its better to code this using NOT EQUAL (e.g. OPERATION_MODE ~= 'chemostat') so it's generalizable to other operation modes that aren't explicitly named "turbidostat" or "chemostat". That way custom modes generate all files, even if not used, and only the specific 'turbidostat' and 'chemostat' modes exclude folders. Forum post forthcoming.